File: flutter/amen_client/lib/src/resources/account.dart
import 'package:http/http.dart' as http;
import '../client.dart';

class AccountResource {
  final AmenClient _c;
  AccountResource(this._c);
  Future<Map<String, dynamic>> get() async => (await _c.request('GET', '/account') as Map).cast();
  Future<List<dynamic>> bankAccounts() async => await _c.request('GET', '/account/bank-accounts/') as List;
  Future<Map<String, dynamic>> linkBankAccount(String iban, {http.MultipartFile? proofDocument}) async =>
      (await _c.request('POST', '/account/bank-accounts/', form: {'iban': iban}, files: proofDocument == null ? null : [proofDocument]) as Map).cast();
  Future<void> deleteBankAccount(String id) => _c.request('DELETE', '/account/bank-accounts/$id');
}

File: swift/Sources/AmenClient/Resources/Account.swift
import Foundation
public final class AccountResource {
    let c: AmenClient; init(_ c: AmenClient) { self.c = c }
    public func get() async throws -> Account { try await c.request(.GET, "/account") }
    public func bankAccounts() async throws -> [BankAccount] { try await c.request(.GET, "/account/bank-accounts/") }
    public func linkBankAccount(iban: String, proofDocument: MultipartFile? = nil) async throws -> BankAccount { try await c.request(.POST, "/account/bank-accounts/", form: ["iban": iban], files: proofDocument.map { [$0] } ?? []) }
    public func deleteBankAccount(_ id: String) async throws { let _: Empty = try await c.request(.DELETE, "/account/bank-accounts/\(id)") }
}
